openssl-dynamic/src/main/c/sslcontext.c

```diff
@@ -2259,13 +2264,21 @@ TCN_IMPLEMENT_CALL(void, SSLContext, setPrivateKeyMethod)(TCN_STDARGS, jlong ctx
             return;
         }
 
-        jmethodID signMethod = (*e)->GetMethodID(e, method_class, "sign", "(JI[B)[B");
+        NETTY_JNI_UTIL_PREPEND(staticPackagePrefix, "io/netty/internal/tcnative/ResultCallback;)V", name, error);
+        NETTY_JNI_UTIL_PREPEND("(JI[BL", name, combinedName, error);
+        TCN_REASSIGN(name, combinedName);
```

Comment:
Out of curiosity - do we have an automated test that makes sure we handle shaded classes correctly, so we don't make a mistake?

Reply:
yes we have it in netty as part of the `testsuite-shading`